bindings/matrix-sdk-ffi/src/room_list.rs

```diff
@@ -441,15 +452,44 @@ impl RoomListItem {
         Ok(RoomInfo::new(self.inner.inner_room(), avatar_url, latest_event).await?)
     }
 
-    /// Building a `Room`.
-    ///
-    /// Be careful that building a `Room` builds its entire `Timeline` at the
-    /// same time.
-    async fn full_room(&self) -> Arc<Room> {
-        Arc::new(Room::with_timeline(
-            self.inner.inner_room().clone(),
-            Arc::new(RwLock::new(Some(Timeline::from_arc(self.inner.timeline().await)))),
-        ))
+    /// Building a `Room`. If its internal timeline hasn't been initialized
+    /// it'll fail.
+    async fn full_room(&self) -> Result<Arc<Room>, RoomListError> {
+        if let Some(timeline) = self.inner.timeline() {
+            Ok(Arc::new(Room::with_timeline(
+                self.inner.inner_room().clone(),
+                Arc::new(RwLock::new(Some(Timeline::from_arc(timeline)))),
+            )))
+        } else {
+            Err(RoomListError::TimelineNotInitialized {
+                room_name: self.inner.inner_room().room_id().to_string(),
+            })
+        }
+    }
+
+    /// Checks whether the Room's timeline has been initialized before.
+    fn is_timeline_initialized(&self) -> bool {
+        self.inner.is_timeline_initialized()
+    }
+
+    /// Initializes the timeline for this room using the provided parameters.
+    /// * `event_type_filter` - An optional [`TimelineEventTypeFilter`] to be
+    ///   used to filter timeline events besides the default timeline filter. If
+    ///   `None` is passed, only the default timeline filter will be used.
+    async fn init_timeline(
+        &self,
+        event_type_filter: Option<Arc<TimelineEventTypeFilter>>,
+    ) -> Result<(), RoomListError> {
+        let timeline_builder = self.inner.default_room_timeline_builder();
+        let timeline_builder = if let Some(event_type_filter) = event_type_filter {
+            timeline_builder.event_filter(move |event, room_version_id| {
+                // Always perform the default filter first
+                default_event_filter(event, room_version_id) && event_type_filter.filter(event)
+            })
+        } else {
+            timeline_builder.event_filter(default_event_filter)
```

Comment:
In this case, you don't have to call that, since it's the default filter, it's preset in the timeline builder default parameters:

https://github.com/matrix-org/matrix-rust-sdk/blob/6e3892528cfc81cc7a56350835e7629e967670fa/crates/matrix-sdk-ui/src/timeline/inner/mod.rs#L128